pkg/tfbridge/provider.go

```diff
@@ -678,6 +679,39 @@ func (p *Provider) Check(ctx context.Context, req *pulumirpc.CheckRequest) (*pul
 	return &pulumirpc.CheckResponse{Inputs: minputs, Failures: failures}, nil
 }
 
+// For properties with MaxItemsOne, where the state is still an array
+// (i.e. from a previous version without MaxItemsOne)
+// we need to mark them for update manually in order to correct the state
+// from an array to a flat type.
+func markWronglyTypedStateDiff(res Resource, olds resource.PropertyMap, diff *shim.InstanceDiff) error {
```

Comment:
Consider never returning an error here. You can return a flag that you detected changes.